src/charm.py

```diff
@@ -1,104 +1,126 @@
 from ops.main import main
-from ops.model import ActiveStatus
+from ops.model import ActiveStatus, MaintenanceStatus
+from ops.pebble import Layer
 
 logger = logging.getLogger(__name__)
 
 
-class OperatorTemplateCharm(CharmBase):
-    """Charm the service."""
-
-    _stored = StoredState()
+class MysqlOperatorCharm(CharmBase):
+    """A Juju Charm for MySQL InnoDB Cluster."""
 
     def __init__(self, *args):
         super().__init__(*args)
-        self.framework.observe(self.on.httpbin_pebble_ready, self._on_httpbin_pebble_ready)
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
-        self.framework.observe(self.on.fortune_action, self._on_fortune_action)
-        self._stored.set_default(things=[])
 
-    def _on_httpbin_pebble_ready(self, event):
-        """Define and start a workload using the Pebble API.
+        self._config = self.model.config
 
-        TEMPLATE-TODO: change this example to suit your needs.
-        You'll need to specify the right entrypoint and environment
-        configuration for your specific workload. Tip: you can see the
-        standard entrypoint of an existing container using docker inspect
+        self.framework.observe(self.on.start, self._on_start)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        change_events = [
+            self.on.mysql_server_pebble_ready,
+            self.on.mysql_router_pebble_ready,
+        ]
+        for event in change_events:
+            self.framework.observe(event, self._on_pebble_ready)
+
+    # ---- Event handlers
+    def _on_start(self, _):
+        """Placeholder for StartEvent."""
+        pass
 
-        Learn more about Pebble layers at https://github.com/canonical/pebble
-        """
-        # Get a reference the container attribute on the PebbleReadyEvent
+    def _on_config_changed(self, _):
+        """Placeholder for ConfigChangedEvent."""
+        pass
+
+    def _on_pebble_ready(self, event: WorkloadEvent) -> None:
+        """Updates the Pebble configuration layer when changed."""
         container = event.workload
-        # Define an initial Pebble layer configuration
-        pebble_layer = {
-            "summary": "httpbin layer",
-            "description": "pebble config layer for httpbin",
+        if not container.can_connect():
+            self.unit.status = MaintenanceStatus("Waiting for pod startup to complete")
+            return
+
+        # Get current config
+        current_layer = container.get_plan()
+        # Create a new config layer
+        service = container.name
+        new_layer = self._mysql_layer(service=service)
+
+        if current_layer.services != new_layer.services:
+            container.add_layer(service, new_layer, combine=True)
+            logging.info("Pebble plan updated with new configuration")
+            if not container.get_service(service).is_running():
+                container.restart(service)
```

Comment:
Maybe this could be replaced by a `container.start`?

Reply:
Probably the condition is not even needed and I can just leave the `container.restart()`.

Reply:
I agree.

Reply:
Just changed it, thanks.